tests/integration/externalapi/test_venafiapi.py

```diff
@@ -25,70 +34,123 @@ def with_random_name(func):
     '''
     generate a randomized name for a container
     '''
+
     @functools.wraps(func)
     def wrapper(self, *args, **kwargs):
         name = _random_name(prefix='salt_')
-        return func(self, _random_name(prefix='salt_test_'), *args, **kwargs)
+        return func(self, _random_name(prefix='salt-test-'), *args, **kwargs)
+
     return wrapper
 
 
-@destructiveTest
-@expensiveTest
+# @destructiveTest
+# @expensiveTest
 class VenafiTest(ShellCase):
     '''
     Test the venafi runner
     '''
 
     @with_random_name
-    def test_gen_key_password(self, name):
-        '''
-        venafi.gen_key
-        '''
-        ret = self.run_run_plus(fun='venafi.gen_key',
-                                minion_id='{0}.test.saltstack.com'.format(name),
-                                dns_name='{0}.test.saltstack.com'.format(name),
-                                zone='Internet',
-                                password='SecretSauce')
-        self.assertEqual(ret['out'][0], '-----BEGIN RSA PRIVATE KEY-----')
-        self.assertEqual(ret['out'][1], 'Proc-Type: 4,ENCRYPTED')
-        self.assertEqual(ret['out'][-1], '-----END RSA PRIVATE KEY-----')
+    def test_request(self, name):
+        print("Testing Venafi request cert")
+        print("Using venafi config:", self.master_opts['venafi'])
+        cn = '{0}.example.com'.format(name)
+        ret = self.run_run_plus(fun='venafi.request',
+                                minion_id=cn,
+                                dns_name=cn,
+                                key_password='secretPassword',
+                                zone=environ.get('CLOUDZONE'))
+        print("Ret is:\n", ret)
+        cert_output = ret['return'][0]
+        if not cert_output:
+            pytest.fail('venafi_certificate not found in output_value')
```

Comment:
This fail is probably not necessary. If `cert_output` is None, `.encode` will fail. If it's not a valid certificate, the load call will fail with a `ValueError`. I would probably just write this as:

```
cert_output = ret['return'][0]
cert = x509.load_pem_x509_certificate(cert_output.encode(), default_backend())
assert cert.subject.get_attributes_for_oid(...) = [...]
```

Reply:
Maybe keep it? Because it easier to understand cause of failure.

Reply:
Changing it to something like this would work:

```
    assert cert_output is not None, 'venafi_certificate not found in `output_value`'
```

Reply:
Ok